Xamarin.Forms.Core/RefreshView.cs

```diff
@@ -20,7 +22,41 @@ public RefreshView()
 		}
 
 		public static readonly BindableProperty IsRefreshingProperty =
-			BindableProperty.Create(nameof(IsRefreshing), typeof(bool), typeof(RefreshView), false, BindingMode.TwoWay);
+			BindableProperty.Create(nameof(IsRefreshing), typeof(bool), typeof(RefreshView), false, BindingMode.TwoWay, coerceValue:OnCoerceIsRefreshing, propertyChanged: OnIsRefreshingPropertyChanged);
+
+		static void OnIsRefreshingPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			bool value = (bool)newValue;
+
+			if (value)
+			{
+				var resfrehView = ((RefreshView)bindable);
+				resfrehView.Refreshing?.Invoke(bindable, EventArgs.Empty);
+				if(resfrehView.Command != null)
+					resfrehView.Command.Execute(resfrehView.CommandParameter);
+			}
+		}
+
+		static object OnCoerceIsRefreshing(BindableObject bindable, object value)
+		{
+			RefreshView view = (RefreshView)bindable;
+			bool newValue = (bool)value;
+
+			// IsRefreshing can always be toggled to false
+			if (!newValue)
+				return value;
+
+			if (!view.IsEnabled)
+				return false;
+
+			if (view.Command == null)
+				return value;
+
+			if (!view.Command.CanExecute(view.CommandParameter))
+				return false;
```

Comment:
Can we group all these if checks together? There's so many of them. Group !newValue and view.Command together. Group !view.IsEnabled and !view.Command.CanExecute() together? Not sure if it'd be more difficult to read this way. Just a thought.

Reply:
hehe
I'll leave it for now unless it gets another down vote.  Helps me think through it better having it split out. Especially knowing how slow future Shane is.